File: bin/system.lua
local uv = pcall(function ()
  return require("uv")
end) and require("uv")
local function call_with_file(f, path, mode)
  local h,e = io.open(path, mode)
  if not h then
    error(e)
  end
  local __x = f(h)
  h.close(h)
  return __x
end
local function read_file(path)
  return call_with_file(function (f)
    return f.read(f, "*a")
  end, path)
end
local function write_file(path, data)
  return call_with_file(function (f)
    return f.write(f, data)
  end, path, "w")
end
local function file_exists63(path)
  local __f = io.open(path)
  local __id = is63(__f)
  local __e
  if __id then
    local __r7 = is63(__f.read(__f, 0)) or 0 == __f.seek(__f, "end")
    __f.close(__f)
    __e = __r7
  else
    __e = __id
  end
  return __e
end
local function directory_exists63(path)
  local __f1 = io.open(path)
  local __id1 = is63(__f1)
  local __e1
  if __id1 then
    local __r9 = not __f1.read(__f1, 0) and not( 0 == __f1.seek(__f1, "end"))
    __f1.close(__f1)
    __e1 = __r9
  else
    __e1 = __id1
  end
  return __e1
end
local path_separator = char(_G.package.config, 0)
local function path_join(...)
  local __parts = unstash({...})
  return reduce(function (x, y)
    return x .. path_separator .. y
  end, __parts) or ""
end
local function get_environment_variable(name)
  return os.getenv(name)
end
local function write(x)
  if uv then
    return uv.write(process.stdout.handle, x)
  else
    return io.write(x)
  end
end
local function exit(code)
  return os.exit(code)
end
local __id2 = arg
local __e3
if __id2 then
  __e3 = __id2
else
  local __e4
  if args then
    __e4 = cut(args, 1)
  end
  __e3 = __e4
end
local argv = __e3
local function reload(module)
  package.loaded[module] = nil
  return require(module)
end
local function shell(command)
  local __f2 = io.popen(command)
  local __x2 = __f2.read(__f2, "*all")
  __f2.close(__f2)
  return __x2
end
return {["read-file"] = read_file, ["write-file"] = write_file, ["file-exists?"] = file_exists63, ["directory-exists?"] = directory_exists63, ["path-separator"] = path_separator, ["path-join"] = path_join, ["get-environment-variable"] = get_environment_variable, write = write, exit = exit, argv = argv, reload = reload, shell = shell}

File: bin/system.js
var fs = require("fs");
var child_process = require("child_process");
read_file = function (path) {
  return fs.readFileSync(path, "utf8");
};
_G.read_file = read_file;
write_file = function (path, data) {
  return fs.writeFileSync(path, data, "utf8");
};
_G.write_file = write_file;
file_exists63 = function (path) {
  return fs.existsSync(path, "utf8") && fs.statSync(path).isFile();
};
_G.file_exists63 = file_exists63;
directory_exists63 = function (path) {
  return fs.existsSync(path, "utf8") && fs.statSync(path).isDirectory();
};
_G.directory_exists63 = directory_exists63;
path_separator = require("path").sep;
_G.path_separator = path_separator;
path_join = function () {
  var __parts = unstash(Array.prototype.slice.call(arguments, 0));
  return reduce(function (x, y) {
    return x + path_separator + y;
  }, __parts) || "";
};
_G.path_join = path_join;
get_environment_variable = function (name) {
  return process.env[name];
};
_G.get_environment_variable = get_environment_variable;
write = function (x) {
  var __out = process.stdout;
  return __out.write(x);
};
_G.write = write;
exit = function (code) {
  return process.exit(code);
};
_G.exit = exit;
argv = cut(process.argv, 2);
_G.argv = argv;
reload = function (module) {
  delete require.cache[require.resolve(module)];
  return require(module);
};
_G.reload = reload;
run = function (command) {
  return child_process.execSync(command).toString();
};
_G.run = run;
exports["read-file"] = read_file;
exports["write-file"] = write_file;
exports["file-exists?"] = file_exists63;
exports["directory-exists?"] = directory_exists63;
exports["path-separator"] = path_separator;
exports["path-join"] = path_join;
exports["get-environment-variable"] = get_environment_variable;
exports.write = write;
exports.exit = exit;
exports.argv = argv;
exports.reload = reload;
exports.run = run;
